test-data/operatorpolicy.json

```diff
@@ -0,0 +1,225 @@
+  },
+  "status": {
+    "compliant": "NonCompliant",
+    "conditions": [
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "CatalogSource was found",
+        "reason": "CatalogSourcesFound",
+        "status": "False",
+        "type": "CatalogSourcesUnhealthy"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "the ClusterServiceVersion required by the policy was not found",
+        "reason": "ClusterServiceVersionMissing",
+        "status": "False",
+        "type": "ClusterServiceVersionCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "NonCompliant; the policy spec is valid, the policy does not specify an OperatorGroup but one already exists in the namespace - assuming that OperatorGroup is correct, the Subscription required by the policy was not found, there are no relevant InstallPlans in the namespace, the ClusterServiceVersion required by the policy was not found, no CRDs were found for the operator, there are no relevant deployments because the ClusterServiceVersion is missing, CatalogSource was found",
+        "reason": "NonCompliant",
+        "status": "False",
+        "type": "Compliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "no CRDs were found for the operator",
+        "reason": "RelevantCRDNotFound",
+        "status": "True",
+        "type": "CustomResourceDefinitionCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "there are no relevant deployments because the ClusterServiceVersion is missing",
+        "reason": "NoRelevantDeployments",
+        "status": "True",
+        "type": "DeploymentCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "an InstallPlan to update to [strimzi-cluster-operator.v0.36.0] is available",
+        "reason": "InstallPlanRequiresApproval",
+        "status": "True",
+        "type": "InstallPlanCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "the policy does not specify an OperatorGroup but one already exists in the namespace - assuming that OperatorGroup is correct",
+        "reason": "PreexistingOperatorGroupFound",
+        "status": "True",
+        "type": "OperatorGroupCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "the Subscription required by the policy was not found",
+        "reason": "SubscriptionMissing",
+        "status": "False",
+        "type": "SubscriptionCompliant"
+      },
+      {
+        "lastTransitionTime": "2024-08-15T16:58:38Z",
+        "message": "the policy spec is valid",
+        "reason": "PolicyValidated",
+        "status": "True",
+        "type": "ValidPolicySpec"
+      }
+    ],
+    "observedGeneration": 1,
+    "relatedObjects": [
```

Comment:
All these `relatedObjects` could be easily stored and shown by search. This is likely out of scope, but wanted to mention it now that we are looking at it. FYI @mprahl

Reply:
Good point. That would be a nice additional feature.

Reply:
Just FYI, ConfigurationPolicy and OperatorPolicy use this same schema, but it is different than some OpenShift `relatedObjects`, which use [ObjectReference](https://github.com/openshift/api/blob/2c10e5887729/config/v1/types_cluster_operator.go#L96) (this caused some problems in must-gather for a while)